### PR TITLE
[Test][Cephadm] Add test to validate mon stack auth

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
Adding DNM, as review is required on the approach

File: tests/cephadm/test_mon_stack_auth.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    # Check if the value is update to true
+    if "false" in CephAdm(installer).ceph.config.get(
+        "mgr", "mgr mgr/cephadm/secure_monitoring_stack"
+    ):
+        raise OperationFailedError(
+            "Failed to set secure_monitoring_stack value to true"
+        )
+
+    # Get dashboard api host
+    dashboard = CephAdm(installer).ceph.dashboard(cmd="get-alertmanager-api-host")
```
Comment (pranavprakash20) [inline tests/cephadm/test_mon_stack_auth.py:44]: instead of cmd, pass as kwargs and use build_cmd_from_args in ceph.py

File: cli/ceph/ceph.py [hunk 2]
```diff
@@ -80,3 +80,17 @@ def health(self, detail=False):
         if isinstance(out, tuple):
             return out[0].strip()
         return out
+
+    def dashboard(self, **kw):
+        """
+        Executed ceph dashboard commands
+        Args:
+            kw (dict): Key/value pairs of dashboard command arg
```
Comment (vamahaja) [inline cli/ceph/ceph.py:88]: if possible add supported keys

File: tests/cephadm/test_mon_stack_auth.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    """Verify redeploy for a specific service
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config", {})
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Get mgr services
+    out = CephAdm(installer).ceph.mgr.services()
+    username = config.get("username")
```
Comment (vamahaja) [inline tests/cephadm/test_mon_stack_auth.py:24]: Add separate comment for these configs

File: tests/cephadm/test_mon_stack_auth.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    password = config.get("password")
+
+    for mon_stack in ["dashboard", "prometheus"]:
```
Comment (vamahaja) [inline tests/cephadm/test_mon_stack_auth.py:27]: Better to have separate methods to check if these services are enabled in utility

File: tests/cephadm/test_mon_stack_auth.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    # Check if the value is update to true
+    if "false" in CephAdm(installer).ceph.config.get(
+        "mgr", "mgr mgr/cephadm/secure_monitoring_stack"
+    ):
+        raise OperationFailedError(
+            "Failed to set secure_monitoring_stack value to true"
+        )
+
+    # Get dashboard api host
+    dashboard = CephAdm(installer).ceph.dashboard({"get-alertmanager-api-host": ""})
```
Comment (vamahaja) [inline tests/cephadm/test_mon_stack_auth.py:53]: Why key is "" ?

File: tests/cephadm/test_mon_stack_auth.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+    # Get prometheus api host
+    prometheus = CephAdm(installer).ceph.dashboard({"get-prometheus-api-host": ""})
+
+    # Try to access the urls without creds
+    for mon_stack in [dashboard, prometheus]:
+        # status_code = "400"
+        ret = Auth(mon_stack, HEADER).check("test", "pwd")
+        log.info(ret)
```
Comment (vamahaja) [inline tests/cephadm/test_mon_stack_auth.py:62]: insufficient data

File: tests/cephadm/test_mon_stack_auth.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    # Now access the urls with creds
+    for mon_stack in [dashboard, prometheus]:
+        # status_code = "200"
+        ret = Auth(mon_stack, HEADER).check(username, password)
+        log.info(ret)
```
Comment (vamahaja) [inline tests/cephadm/test_mon_stack_auth.py:68]: insufficient data

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/cephadm/test_mon_stack_auth.py (hunk 4 above)
Comment (pranavprakash20) [reply tests/cephadm/test_mon_stack_auth.py:27]: added

File: cli/ceph/ceph.py (hunk 2 above)
Comment (pranavprakash20) [reply cli/ceph/ceph.py:88]: added

**Comment by pranavprakash20** [comment]
run log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-A0SYAL/

File: cli/ops/cephadm.py [hunk 8]
```diff
@@ -182,3 +182,26 @@ def bootstrap(
...
+
+
+def get_mon_stack_status(installer, mon_stack):
+    """Get dashboard status
+    Args:
+        installer (CephInstallerNode): Ceph installer node
+        mon_stack (str): Monitoring stack name whose status needs to be checked
+    """
+    out = CephAdm(installer).ceph.mgr.services()
+    if mon_stack not in out:
```
Comment (vamahaja) [inline cli/ops/cephadm.py:205]: better to regex and return the status

File: cli/ops/cephadm.py [hunk 9]
```diff
@@ -182,3 +182,26 @@ def bootstrap(
         raise OperationFailedError("Failed to bootstrap cluster")
 
     return True
+
+
+def is_mon_stack_enabled(installer, mon_stack):
```
Comment (vamahaja) [inline cli/ops/cephadm.py:187]: instead general method, create methods specific to mon stacks like `is_prometheus_enabled`

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
run log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-12885S/

**Review by sayaleeraut — APPROVED** [review]
Changes LGTM!

File: cli/ops/cephadm.py [hunk 10]
```diff
@@ -182,3 +184,35 @@ def bootstrap(
...
+    """Checks if dashboard is enabled
+    Args:
+        installer (CephInstallerNode): Ceph installer node
+    """
+    if not get_mon_stack_status(installer, "dashboard") == "on":
+        return False
+    return True
+
+
+def get_mon_stack_status(installer, mon_stack):
```
Comment (vamahaja) [inline cli/ops/cephadm.py:209]: Change this to `get_module_status`, update veriables accordingly

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: cli/ops/cephadm.py (hunk 10 above)
Comment (pranavprakash20) [reply cli/ops/cephadm.py]: Updated.

**Review by tintumathew10 — COMMENTED** [review]
changes looks good

**Review by vamahaja — APPROVED** [review]

**Review by sayaleeraut — APPROVED** [review]
LGTM!

**Pull request merged**